src/infiniteLoops/runtime.ts

```diff
@@ -279,10 +280,18 @@ export function testForInfiniteLoop(code: string, previousCodeStack: string[]) {
 
   const state = new st.State()
 
-  const sandboxedRun = new Function('code', functionsId, stateId, builtinsId, 'return eval(code)')
+  const sandboxedRun = new Function(
+    'code',
+    functionsId,
+    stateId,
+    builtinsId,
+    MODULE_PARAMS_ID,
+    // redeclare window so modules don't do anything funny like play sounds
+    '{let window = {}; return eval(code)}'
```

Comment:
Have you tested this in browser? I'm not convinced it'll work, actually. A permanent fix would probably need a change in the modules side.

It's good enough for now, though. If something is timing out, their code will probably be stuck before any side effects occur.

Reply:
yup I tested in the browser

I ran something like
```import {play, sine_sound} from 'sound';
play(sine_sound(1200,1));
function f(x){f(x);}
f(1);
```
and the sound only played once. The infinite loop wasn't detected though since `play` threw an error first (in the instrumented code)

Reply:
Hmm on closer inspection, it might actually work, I misread some parts of the module loading process.

Ignore this, then.